Reorganize/edit/update some sections (#1069)
composewell/streamly: README.md

* Powerful abstractions
* Declarative concurrency

This guide introduces you to streamly using some

> practical examples to see it working. You can find the

Reviewer: ```suggestion
practical examples. You can find the
```

Author: ok.

> working code of these examples in the [streamly-examples

Reviewer: Will this link be rendered properly?

Author: Yes. See https://github.com/composewell/streamly/tree/readme .

  & Stream.fold Fold.length  -- IO Int
```

If you are wondering about the `Internal` module import, see [A Note on

> Internal Modules](#a-not-on-internal-modules) section.

Reviewer: Broken link. Should be `#a-note-on-internal-modules`.

```suggestion
Please go through [this](#a-note-on-internal-modules) section 
for more details on internal modules.
```

Author: fixed already!

then we recommend that you use the plugin for performance sensitive
applications.

## A Note on Internal Modules

You may see some `Internal` modules imported in the examples. Those are

> not really Internal to the library.  We can classify streamly modules in

Reviewer: ```suggestion
not really Internal to the library.  We can classify streamly modules into
```

Author: ok.

two categories:

> * Released Modules: these are modules that have a stable API, any API changes

Reviewer: ```suggestion
* Released Modules: These are modules that have a stable API, any API changes
```

Author: ok.

>   conform to a versioning policy.

Reviewer: ```suggestion
  conform to the versioning policy.
```

Author: "the package versioning policy".

* Pre-release Modules: These modules are not yet released due to some planned
  changes in near future, they will be released soon. These modules reside in

>   the `Internal` namespace for now.

Reviewer: I'm unable to phrase this properly. But replace `some planned changes in near future` with something regarding having a soak time for the API.

## Contributing

* [Streamly Contributor's Guide](CONTRIBUTING.md)
* [Contact the streamly development team](mailto:streamly@composewell.com)

## License

Streamly is an [open source](https://github.com/composewell/streamly)
project available under [BSD-3-Clause
license](https://github.com/composewell/streamly/blob/master/LICENSE).

## Support

* Join streamly [online community chat](https://gitter.im/composewell/streamly)
* For professional support please contact [support@composewell.com](mailto:support@composewell.com)

> # [Streamly Homepage](https://streamly.composewell.com)

Reviewer: This can be removed.

Author: Need to have the homepage link at some prominent place. I add a reference in the intro para then.
